modules/reactor/src/vectorpostprocessors/IntegralReportingIDVPP.C

```diff
@@ -0,0 +1,180 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/*  MAMMOTH: - Radiation Kinetics and Reactor Physics Software  */
+/*                                                              */
+/*           (c) 2016 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "IntegralReportingIDVPP.h"
+
+#include "MooseVariable.h"
+#include "MooseMesh.h"
+// #include "MyMath.h"
+
+#include "libmesh/mesh_base.h"
+
+registerMooseObject("ReactorApp", IntegralReportingIDVPP);
+
+InputParameters
+IntegralReportingIDVPP::validParams()
+{
+  InputParameters params = ElementVariableVectorPostprocessor::validParams();
+  params.addRequiredParam<std::vector<ExtraElementIDName>>(
+      "reporting_id", "The types of reporting id by which to separate integrals.");
+  params.addClassDescription(
+      "This IntegralReportingIDVPP source code is to integrate variables based on reporting IDs");
+  return params;
+}
+
+IntegralReportingIDVPP::IntegralReportingIDVPP(const InputParameters & parameters)
+  : ElementVariableVectorPostprocessor(parameters),
+    _nvar(coupledComponents("variable")),
+    _reporting_id(getParam<std::vector<ExtraElementIDName>>("reporting_id")),
+    _n_reporting_id(_reporting_id.size())
+{
+  for (auto id_type : _reporting_id)
+  {
+    unsigned int id_index = getElementIDIndexByName(id_type);
+    std::set<dof_id_type> ids;
+    if (blockRestricted())
+    {
+      ids = getElemIDsOnBlocks(id_index, blockIDs());
+    }
+    else
+    {
+      ids = getAllElemIDs(id_index);
+    }
+    _reporting_id_index.push_back(id_index);
+    std::map<dof_id_type, dof_id_type> id_map;
+    std::map<dof_id_type, dof_id_type> id_inverse_map;
```

Comment:
I think the data types can be improved here

also I d be surprised if none of these two maps don't exist already. Check libmesh doxygen just in case

Reply:
Updated this VPP not to use those two data types.